src/components/Environment/environment-utils.ts

```diff
@@ -58,3 +95,50 @@ export const createEnvironmentKindFromReleasePlan = (
     deploymentStrategy: EnvironmentDeploymentStrategy.Manual,
   },
 });
+
+const createCredentialsSecret = async (kubeconfig: string, namespace: string, dryRun = false) => {
+  const resource: SecretKind = {
+    apiVersion: SecretModel.apiVersion,
+    kind: SecretModel.kind,
+    metadata: { generateName: 'env-cluster-credentials-', namespace },
+    type: 'managed-gitops.redhat.com/managed-environment',
+    stringData: { kubeconfig },
+  };
+  return k8sCreateResource({
+    model: SecretModel,
+    resource,
+    ...(dryRun && { queryOptions: { queryParams: { dryRun: 'All' } } }),
+  });
+};
+
+export const createEnvironment = async (
+  values: CreateEnvironmentFormValues,
+  namespace: string,
+  dryRun = false,
+) => {
+  const secret = await createCredentialsSecret(values.kubeconfig, namespace, dryRun);
+  const kubeconfig = YAML.load(values.kubeconfig) as KubeConfig;
+  const resource: EnvironmentKind = {
+    apiVersion: `${EnvironmentModel.apiGroup}/${EnvironmentModel.apiVersion}`,
+    kind: EnvironmentModel.kind,
+    metadata: { name: values.name.toLowerCase().trim().split(' ').join('-'), namespace },
+    spec: {
+      displayName: values.name,
+      deploymentStrategy: EnvironmentDeploymentStrategy[values.deploymentStrategy],
+      tags: ['managed'],
```

Comment:
Maybe add this tag based on the selection of dropdown for BYOC or static. Right now it will only have BYOC option but later it can have static as well.

Reply:
Ok, makes sense. Updated.